google/resource_google_project_default_service_accounts_test.go

```diff
@@ -0,0 +1,287 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	cloudresourcemanager "google.golang.org/api/cloudresourcemanager/v1"
+)
+
+func TestAccResourceGoogleProjectDefaultServiceAccountsBasic(t *testing.T) {
+	t.Parallel()
+
+	resourceName := "google_project_default_service_accounts.acceptance"
+	org := getTestOrgFromEnv(t)
+	project := fmt.Sprintf("tf-project-%d", randInt(t))
+	billingAccount := getTestBillingAccountFromEnv(t)
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckGoogleProjectDefaultServiceAccountsBasic(org, project, billingAccount),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "id", "projects/"+project),
+					resource.TestCheckResourceAttrSet(resourceName, "project"),
+					resource.TestCheckResourceAttrSet(resourceName, "action"),
+					resource.TestCheckResourceAttrSet(resourceName, "project"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleProjectDefaultServiceAccountsBasic(org, project, billingAccount string) string {
+	return fmt.Sprintf(`
+resource "google_project" "acceptance" {
+	project_id = "%s"
+	name       = "%s"
+	org_id  = "%s"
+	billing_account = "%s"
+}
+
+resource "google_project_default_service_accounts" "acceptance" {
```

Comment:
Looking at this resource, I as a user don't really know what it's intending to accomplish because the behavior is all in default values. I wonder whether it actually makes sense to make some fields required instead of default so that the user's intent is more explicit. What do you think?

Reply:
agreed